lib/ash/resource/validation/changing.ex

```diff
@@ -53,16 +53,42 @@ defmodule Ash.Resource.Validation.Changing do
   end
 
   @impl true
-  def atomic(_changeset, opts, context) do
-    {:atomic, [opts[:field]], expr(^atomic_ref(opts[:attribute]) == ^ref(opts[:attribute])),
-     expr(
-       error(^InvalidAttribute, %{
-         field: ^opts[:field],
-         value: ^atomic_ref(opts[:attribute]),
-         message: ^(context.message || "must be changing"),
-         vars: %{field: ^opts[:field]}
-       })
-     )}
+  def atomic(changeset, opts, context) do
+    case Ash.Resource.Info.relationship(changeset.resource, opts[:field]) do
+      nil ->
+        if not Ash.Changeset.changing_attribute?(changeset, opts[:field]) do
+          {:error, exception(opts)}
+        else
+          {:atomic, [opts[:field]], expr(^atomic_ref(opts[:field]) == ^ref(opts[:field])),
+           expr(
+             error(^InvalidAttribute, %{
+               field: ^opts[:field],
+               value: ^atomic_ref(opts[:field]),
+               message: ^(context.message || "must be changing"),
+               vars: %{field: ^opts[:field]}
+             })
+           )}
+        end
+
+      %{type: :belongs_to, source_attribute: source_attribute} ->
+        if not Ash.Changeset.changing_attribute?(changeset, source_attribute) do
```

Comment:
lets flip the conditional (I think credo will yell at you)

Reply:
Done